From: Simon McVittie <simon.mcvittie@collabora.co.uk>
Date: Mon, 9 Mar 2015 09:48:16 +0000
Subject: power: make sure to set an error when GDBus set_property fails

If we return FALSE, the API is that the error must be set, and
GDBus asserts this.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=745920
Bug-Debian: https://bugs.debian.org/775877
Reviewed-by: Bastien Nocera
Applied-upstream: 3.15.92, commit:4e8448a; 3.14.3, commit:d91542c
---
 plugins/power/gsd-power-manager.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index b31ec6d..eafa082 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -2771,6 +2771,8 @@ handle_set_property (GDBusConnection *connection,
         /* Check session pointer as a proxy for whether the manager is in the
            start or stop state */
         if (manager->priv->session == NULL) {
+                g_set_error (error, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                             "Manager is starting or stopping");
                 return FALSE;
         }
 
